apps/client/src/common/components/timer-display/TimerDisplay.tsx

```diff
@@ -19,12 +19,18 @@ interface TimerDisplayProps {
 const TimerDisplay = (props: TimerDisplayProps) => {
   const { time, small, hideZeroHours, className = '' } = props;
 
-  const display =
-    (time === null || typeof time === 'undefined' || isNaN(time))
-      ? '-- : -- : --'
-      : formatDisplay(millisToSeconds(time), hideZeroHours);
+  let display = '';
+
+  if (typeof time === 'string') {
+    display = time;
+  } else if (time === null || typeof time === 'undefined' || isNaN(time)) {
+    display = '-- : -- : --';
+  } else {
+    display = formatDisplay(millisToSeconds(time), hideZeroHours);
+  }
```

Comment:
this logic seems to be repeated in TimerDisplay, Timer and MinimalTimer.
Could we make it a function in the viewerUtils?